scripts/azureml-assets/azureml/assets/copy_assets.py

```diff
@@ -31,9 +35,31 @@ def copy_asset(asset_config: assets.AssetConfig,
         # Skip a released version
         return None
 
+    if check_previous_release:
+        # Skip if previous version was not released
+        previous_release_version = assets.get_latest_release_tag_version(asset_config, release_directory_root)
+
+        if previous_release_version is not None:
+            image = asset_config.extra_config_as_object().get_full_image_name()
+
+            # Check against MCR by making a manifest call to see if tag exists
+            (hostname, repo) = image.split("/", 1)
+            encoded_tag = urllib.parse.quote(previous_release_version, safe="")
+
+            request = Request(f"https://{hostname}/v2/{repo}/manifests/{encoded_tag}",
+                                method="HEAD",
+                                headers={'Accept': "application/vnd.docker.distribution.manifest.v2+json"})
+
+            try:
+                response = assets._urlopen_with_retries(request)
+            except Exception as e:
+                logger.log_error(f"Image {image} does not exist in MCR")

```

Comment:
Should this logging only be for when we get a 404?

Reply:
Yes. If there's some other exception we'd probably want to log that too, but not say that any error means the image doesn't exist.